djangocms_versioning_filer/monkeypatch/admin/clipboardadmin.py

```diff
@@ -61,9 +66,28 @@ def ajax_upload(request, folder_id=None):
         if uploadform.is_valid():
             file_obj = uploadform.save(commit=False)
             # Enforce the FILER_IS_PUBLIC_DEFAULT
-            file_obj.is_public = filer_settings.FILER_IS_PUBLIC_DEFAULT
-            file_obj.folder = folder
+            file_obj.iNos_public = filer_settings.FILER_IS_PUBLIC_DEFAULT
 
+            # manage folder allocations
+            path = request.POST.get('path')
+            # import ipdb; ipdb.set_trace()
+            if path:
+                pathSplit = path.split('/')
+                # if no folders in supplied path, then add to current folder
+                if len(pathSplit) == 0:
+                    file_obj.folder = folder
+                # if there are folders in the supplied path, find / create as necessary
+                else:
+                    subfolder = folder
+                    for segment in pathSplit:
+                        subfolder = check_folder_exists_in_folder(subfolder, segment)
+                        if subfolder:
+                            folder = subfolder
```

Comment:
I find this part of the code really difficult to read/understand - `subfolder = folder` and then `folder = subfolder` is quite confusing. Wondering if it wouldn't be a bit simpler if some of this logic was taken out into a helper function. Otherwise make sure to comment this section of code well cause it's going to be difficult for any dev who has to work in this in the future.

Reply:
Also, separating this out into a helper function would allow you to write a unit test for it (which I think would be a good idea - the code is complicated enough that a unit test would definitely help).